tests/test_asynccog.py

```diff
@@ -0,0 +1,404 @@
+"""test AsyncTilerFactory."""
+
+from io import BytesIO
+from unittest.mock import patch
+
+import numpy
+import pytest
+from rio_tiler.io import COGReader
+from rio_viz.compat import AsyncReader
+
+from titiler.endpoints import factory
+
+from .conftest import mock_rasterio_open, parse_img
+
+from fastapi import FastAPI
+
+from starlette.testclient import TestClient
+
+AsyncCOGReader = type("AsyncReader", (AsyncReader,), {"reader": COGReader})
```

Comment:
@geospatial-jeff not sure if you'll have time, but while doing your aiocogeo (async) vs rio-tiler (sync) benchmark it would be nice to also benchmark async rio-tiler using `rio-viz` `AsyncReader` compat class.

```python
from rio_tiler.io import COGReader
from rio_viz.compat import AsyncReader

AsyncCOGReader = type("AsyncReader", (AsyncReader,), {"reader": COGReader})

async with AsyncCOGReader("my.tif") as cog:
     img = await cog.tile(....)
```